source/agora/network/Manager.d

```diff
@@ -905,18 +905,21 @@ public class NetworkManager
                     blocks[0].header.height, blocks[$ - 1].header.height);
 
                 // one or more blocks were rejected, stop retrieval from node
-                if (!onReceivedBlocks(blocks, preimages))
-                    continue LNextNode;
-
-                block_height += blocks.length;
+                try
+                {
+                    block_height = onReceivedBlocks(blocks, preimages);
+                    if (block_height == 0)
+                        continue LNextNode;
```

Comment:
The special casing on `0` seems odd. Why not return the number of blocks added ?

Reply:
Right I seem to have missed this. Will fix.

Reply:
In fact I can get rid of the do/while loop now too, because I made the function nothrow.